### PR TITLE
[feat] support deta in adapter-auto using adapter-node

### PR DESCRIPTION
**Pull request by ak4zh**
Supports deployment to `deta.space` with `adapter-auto`.
Internally it uses `adapter-node` which is what `deta.space` requires.

Currently, sveltekit apps can be successfully deployed to `deta` by setting the adapter to `adapter-node`. But the problem arises when you want to deploy the same codebase to multiple platforms say deta and vercel. After this change, a sveltekit app can be deployed to both deta and vercel both using `adapter-auto`.

### Please don't delete this checklist! Before submitting the PR, please make sure you do the following:
- [x] It's really useful if your PR references an issue where it is discussed ahead of time. In many cases, features are absent for a reason. For large changes, please create an RFC: https://github.com/sveltejs/rfcs
- [x] This message body should clearly illustrate what problems it solves.
- [ ] Ideally, include a test that fails without this PR but passes with it.

### Tests
- [x] Run the tests with `pnpm test` and lint the project with `pnpm lint` and `pnpm check`

### Changesets
- [x] If your PR makes a change that should be noted in one or more packages' changelogs, generate a changeset by running `pnpm changeset` and following the prompts. Changesets that add features should be `minor` and those that fix bugs should be `patch`. Please prefix changeset messages with `feat:`, `fix:`, or `chore:`.

### CODE REVIEW

File: packages/adapter-auto/adapters.js [hunk 1]
```diff
@@ -24,5 +24,11 @@ export const adapters = [
 		test: () => process.env.GITHUB_ACTION_REPOSITORY === 'Azure/static-web-apps-deploy',
 		module: 'svelte-adapter-azure-swa',
 		version: '0.13'
+	},
+	{
+		name: 'Deta',
+		test: () => !!process.env.DETA_PROJECT_KEY,
+		module: 'svelte-adapter-node',
+		version: '1.1.8'
```
Comment (Rich-Harris) [inline packages/adapter-auto/adapters.js:32]:
```suggestion
		module: '@sveltejs/adapter-node',
		version: '1'
```

File: .changeset/silly-dogs-bow.md [hunk 2]
```diff
@@ -0,0 +1,5 @@
+---
+'@sveltejs/adapter-auto': patch
+---
+
+support deta in adapter-auto using adapter-node
```
Comment (Rich-Harris) [inline .changeset/silly-dogs-bow.md:5]:
```suggestion
chore: add zero-config support for deta
```

**Review by Rich-Harris — CHANGES_REQUESTED** [review]
Thank you for the PR, though is this truly zero config? How does Deta know the server's entry point/port/etc?

If it's _not_ zero config, then the app author should specify the adapter in `svelte.config.js` (using environment variables to switch between adapters as appropriate)

File: packages/adapter-auto/adapters.js (hunk 1 above)
Comment (ak4zh) [reply packages/adapter-auto/adapters.js:32]: It seems `deta` is something like a docker deployment which they call a personal cloud.
So each deployment gets the same production port (probably `4173`) which they already handle internally when the adapter-node is used.

Using the environment variable seems like a good alternative in that case we won't need this change.

**Comment by Conduitry** [comment]
How does Deta know what the entrypoint of the application is? How does it avoid deploying dev dependencies as well? I found no actual documentation on their site that I could read without signing up. You mentioned Docker. If deploying properly involves writing a Dockerfile, then this isn't zero-config, and we shouldn't be having the auto adapter do this.

**Comment by ak4zh** [comment]
@Conduitry Deta is serverless. It need the `adapter-node` no other congiduration is requires.
https://deta.space/docs/en/quickstart-guides/sveltekit

**Comment by Rich-Harris** [comment]
Thanks for the clarification. It looks like an app needs a Spacefile which configures an 'engine', which presumably is how it knows to run the server in `build/index.js`.

The need for a Spacefile means that this isn't zero-config, which really disqualifies it from being included in `adapter-auto`, so I'm going to close this. I think the current approach of documenting the use of `adapter-node` is a better way.

**Comment by ak4zh** [comment]
The suggested method above suggests to use environment variable in `svelte.config.js` but it seems `$env/static/private` is not accessible in `svelte.config.js`

How to access the environment variable ?

```js
import adapter from '@sveltejs/adapter-auto';
import { adapter as adapterNode } from '@sveltejs/adapter-auto';
import { IS_DOCKER } from '$env/static/private'

const config = {
	kit: {
		adapter: IS_DOCKER === 'true' ? adapterNode() : adapter()
	}
}
```

I get error:
```bash
Error [ERR_MODULE_NOT_FOUND]: Cannot find package '$env' imported from /app/svelte.config.js
    at new NodeError (node:internal/errors:393:5)
    at packageResolve (node:internal/modules/esm/resolve:860:9)
    at moduleResolve (node:internal/modules/esm/resolve:909:20)
    at defaultResolve (node:internal/modules/esm/resolve:1124:11)
    at nextResolve (node:internal/modules/esm/loader:163:28)
    at ESMLoader.resolve (node:internal/modules/esm/loader:841:30)
    at ESMLoader.getModuleJob (node:internal/modules/esm/loader:424:18)
    at ModuleWrap.<anonymous> (node:internal/modules/esm/module_job:76:40)
    at link (node:internal/modules/esm/module_job:75:36)
```

**Pull request closed without merging**